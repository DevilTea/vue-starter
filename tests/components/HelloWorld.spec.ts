import { shallowMount } from '@vue/test-utils'
import HelloWorld from '@/components/HelloWorld.vue'
import { createI18n } from 'vue-i18n'

test('shallowMount component', async () => {
  expect(HelloWorld).toBeTruthy()

  const wrapper = shallowMount(HelloWorld, {
    global: {
      plugins: [createI18n({})],
    },
    props: {
      msg: 'Welcome to Your Vue.js + TypeScript App',
    },
  })
  expect(wrapper.text()).toContain('Welcome to Your Vue.js + TypeScript App')
  expect(wrapper.html()).toMatchSnapshot()
})
