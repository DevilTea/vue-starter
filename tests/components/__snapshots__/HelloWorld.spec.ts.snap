// Vitest Snapshot v1

exports[`shallowMount component 1`] = `
"<h1 data-v-6171b430=\\"\\">Welcome to Your Vue.js + TypeScript App</h1>
<p data-v-6171b430=\\"\\">hello-i18n</p>
<p data-v-6171b430=\\"\\"> Recommended IDE setup: <a href=\\"https://code.visualstudio.com/\\" target=\\"_blank\\" data-v-6171b430=\\"\\"> VSCode </a> + <a href=\\"https://github.com/johnsoncodehk/volar\\" target=\\"_blank\\" data-v-6171b430=\\"\\"> Volar </a></p>
<p data-v-6171b430=\\"\\">See <code data-v-6171b430=\\"\\">README.md</code> for more information.</p>
<p data-v-6171b430=\\"\\"><a href=\\"https://vitejs.dev/guide/features.html\\" target=\\"_blank\\" data-v-6171b430=\\"\\"> Vite Docs </a> | <a href=\\"https://v3.vuejs.org/\\" target=\\"_blank\\" data-v-6171b430=\\"\\"> Vue 3 Docs </a></p>
<button type=\\"button\\" data-v-6171b430=\\"\\"> count is: 0</button>
<p data-v-6171b430=\\"\\"> Edit <code data-v-6171b430=\\"\\">components/HelloWorld.vue</code> to test hot module replacement. </p>"
`;
